contracts/WithdrawalsReserves.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.4.8;
+
+import "./Owned.sol";
+import "./ServicesI.sol";
+import "./vendor/MathLib.sol";
+import "./vendor/SafeSendLib.sol";
+
+/* Central reserves for all value in flight
+ * includes: requested withdrawals (emitter excess), contract values after
+ *           dissolution, participant funds before match
+ */
+contract WithdrawalsReserves is Owned {
+    using MathLib for uint;
+    using SafeSendLib for address;
+
+    mapping (address => uint) public balances;
+
+    address servicesAddress;
+
+    event Reserve(address participant, uint amount);
+    event Withdraw(address participant, uint amount);
+    event Transfer(address participant, uint amount);
+
+    function WithdrawalsReserves(address _servicesAddress) {
+        servicesAddress = _servicesAddress;
+    }
+
+    modifier fromFactory() {
+        ServicesI services = ServicesI(servicesAddress);
+        address factoryAddress = services.serviceAddress(sha3("Factory"));
+        if (msg.sender != factoryAddress) { throw; }
+        _;
+    }
+
+    function reserve(address _participant) payable {
+        uint amount = msg.value;
+        balances[_participant] = balances[_participant].safeAdd(amount);
+
+        Reserve(_participant, amount);
+    }
+
+    function withdraw() {
+        address participant = msg.sender;
+        uint amount = balances[participant];
+
+        // re-entry protection
+        balances[participant] = 0;
+
+        uint sent = participant.safeSend(amount);
```

Comment:
quoth @chatch:

> Could just use Solidity send() instead of SafeSend. SafeSend is sending more gas then needed (10000 or 30000?) where send() will use just 2300. I think the extra checks it does aren't needed either as your already checking the balance and have previously checked (received from anyway) valid addresses as well, so there is a little more gas used to do these checks.